frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -797,10 +826,19 @@ pub mod pallet {
                 ));
             }
 
-            T::Currency::resolve_creating(&staker, reward_imbalance);
-            Self::update_staker_info(&staker, &contract_id, staker_info);
-            Self::deposit_event(Event::<T>::Reward(staker, contract_id, era, staker_reward));
+            if let Some(delegate) = RewardDelegate::<T>::get(&staker, &contract_id) {
```

Comment:
Can this be solved in a more optimal way, in relationship to weight?

Reply:
Since the delegation or not of the rewards are restricted by the restaking being set or not, I think a read to storage could be avoid in those cases.

Reply:
Sure, that's true.

You also read in the value few lines above, so it would be more optimal to just reuse that.